src/server/controller/user/exportToCsvStream.ts

```diff
@@ -0,0 +1,70 @@
+import { createI18nPromise } from 'i18n/i18nFactory'
+import { Objects } from 'utils/objects'
+
+import { Areas } from 'meta/area'
+import { Lang } from 'meta/lang'
+import { RoleName, User, Users } from 'meta/user'
+
+import { UserRoleAdapter } from 'server/repository/adapter'
+import { usersBuildGetManyQuery, UsersGetManyProps } from 'server/repository/public/user'
+import { ExportService } from 'server/service/export'
+
+type Props = UsersGetManyProps & {
+  lang: Lang
+}
+
+export const exportToCsvStream = async (props: Props): Promise<NodeJS.ReadableStream> => {
+  const { lang } = props
+
+  const { query, queryParams } = usersBuildGetManyQuery(props)
+
+  const i18n = await createI18nPromise(lang)
+  const nameHeader = i18n.t('common.name')
+  const emailHeader = i18n.t('common.email')
+
+  const roleHeaders: Record<RoleName, string> = {
+    [RoleName.ADMINISTRATOR]: i18n.t(Users.getI18nRoleLabelKey(RoleName.ADMINISTRATOR)),
+    [RoleName.REVIEWER]: i18n.t(Users.getI18nRoleLabelKey(RoleName.REVIEWER)),
+    [RoleName.NATIONAL_CORRESPONDENT]: i18n.t(Users.getI18nRoleLabelKey(RoleName.NATIONAL_CORRESPONDENT)),
+    [RoleName.ALTERNATE_NATIONAL_CORRESPONDENT]: i18n.t(
+      Users.getI18nRoleLabelKey(RoleName.ALTERNATE_NATIONAL_CORRESPONDENT)
+    ),
+    [RoleName.COLLABORATOR]: i18n.t(Users.getI18nRoleLabelKey(RoleName.COLLABORATOR)),
+    [RoleName.VIEWER]: i18n.t(Users.getI18nRoleLabelKey(RoleName.VIEWER)),
+  }
+
+  const rowTransformer = (rawUser: User): Record<string, string> => {
```

Comment:
I typed rawUser here as `User` for now, but it would actually be a snake case version of it. If we want to add the snake cased version, an option would be introducing [SnakeCasedProperties](https://github.com/sindresorhus/type-fest/blob/main/source/snake-cased-properties.d.ts) from [type-fest](https://github.com/sindresorhus/type-fest/tree/main):
`type SnakeCaseUser = SnakeCasedProperties<User>;`
and we could re-use it elsewhere. 

(The other 'any' types were removed)